### ISSUE TITLE
Hello from GitHub 👋 

### ISSUE THREAD

**Issue by lukehefson** [sindresorhus/refined-github#1469]
Hey folks! I'm a Product Manager at GitHub and as @sindresorhus [has mentioned](https://github.com/sindresorhus/refined-github/issues/1458#issuecomment-409999543), and many of you in the refined-github community have already noticed, we've [recently built out a team within GitHub](https://twitter.com/lukehefson/status/1022553956466216960) to address many of those UX pain-points that have been surfaced here (along with other popular community-run groups).

I can even let you into a secret 🤫 – internally, we're calling it '[Papercuts](https://en.wikipedia.org/wiki/Paper_cut_bug)' 📄✂️

I emailed Sindre the other day to say hello and let him know that I'd love to engage more with y'all here in the repo itself… and he gave me the 👍.💚:smile:

Sometimes, our implementations might not be exactly the same as refined-github – but at its core we're trying to address the same problems. We'll be adding some of them to our [changelog](http://blog.github.com/changelog/) – but other 'smaller ships' might just drop without fanfare. Unfortunately, due to various moving parts and different development cadences at GitHub, I can't let you know if or when they'll happen in advance, but I'm always hungry to hear feedback ([tweet me](https://twitter.com/lukehefson), or @_mention me here) and I'll be watching this repo in order to try to keep up as things land.

I can't tell you enough how much I appreciate the refined-github community. We're listening 👂 and striving to build a better GitHub! :octocat:

**Comment by waldyrious** [comment]
@lukehefson very happy to see GitHub not only acting in response to community needs, but actually engaging in conversation, which has been lacking in the past.

What can we expect in terms of the expanded efforts of this kind that you mention above? I'm particularly curious about which long-running requests are still possible, which are on the roadmap, and which have been evaluated and decided against. Some comments from a GitHubber at open issues at https://github.com/isaacs/github would be great, for example.

We understand that you might not want to reveal *all* your plans, but we'd really appreciate at least a bit more communication, to avoid staying the dark for years waiting for features that may never come.

**Comment by FezVrasta** [comment]
I think it'd be good to know which of the refined-github features (current and planned) you are sure will **never** get implemented by GitHub directly, so that the contributors of this repository can decide if they want to focus on their development or just wait for GitHub to implement them and use their time for other features that are marked as "never going to be implemented in GitHub".

Just my .2 cents

**Comment by jasoncwarner** [comment]
Hi @waldyrious I can't and won't comment on any one fix or implementation, though I am able to talk about the goal. We aim to do exactly what you said above: engage in a dialog with the community on what we are building or will build in the near future along these lines. The community has spoken loud and clear and we needed to and are listening; it's time for us to engage with you in your spaces. I would encourage you to continue building things and talking about ways to make GitHub better and the folks at GitHub will engage with you all here (and elsewhere too, as you noted). 

Most of the work we are prioritizing right now would be called nits or papercuts, though our internal list includes some much larger and deeper projects as well. 

And @FezVrasta that is great feedback and @lukehefson and we will make sure to be open with that and you. 

Thank you all for working on refined-github! We can't wait to work with you all more directly on making GitHub better and better. Thank you!

**Comment by jens1o** [comment]
I honestly like what you are doing guys from GitHub, but then, these implementations are no longer open-source :z
I'd like them to be open-sourced, though.

**Comment by 734MNOOB** [comment]
Wow. Fantastic example of "big business" engaging with their front line. I really hope this generates some movement on HN and other feeds, as this seems like such a fantastic opportunity for everyone to come together and have their voices heard in some way, but also learn. I'm keen to see which UX aspects Github pick up but more curious around which ones they don't.

@FezVrasta that's a really good idea! Would mean the users get the best of both worlds 👍🏼

**Comment by jamiebuilds** [comment]
Would be nice if you gave shoutouts to the developers who built the feature into refined-github in the first place.

It's pretty easy to find the history of who worked on a feature in this repo. For example:

- https://github.com/sindresorhus/refined-github/pull/254 / https://github.com/sindresorhus/refined-github/pull/282
- https://blog.github.com/changelog/2018-07-31-unselectable-diff-markers/

The people here have effectively been doing free UX research for GitHub. That deserves credit (well it deserves $ but credit is nice too).

**Comment by rarkins** [comment]
It’s great to see these features/fixes getting landed, however GitHub’s approach of “we can never confirm or deny a single feature 
/fix we plan to do no matter how important or trivial” is a really weird cultural thing I hope gets changed one day. It just seems like stubborn secrecy. Of course, I know it’s not this team’s own choice because it extends across the entire company/product.

**Comment by lukehefson** [comment]
> I think it'd be good to know which of the refined-github features (current and planned) you are sure will never get implemented by GitHub directly, so that the contributors of this repository can decide if they want to focus on their development or just wait for GitHub to implement them and use their time for other features that are marked as "never going to be implemented in GitHub".

[…]

> And `@FezVrasta` that is great feedback and `@lukehefson` and we will make sure to be open with that and you.

@FezVrasta Honestly, we don't know any for certain just yet (particularly as _most_ of the ideas within refined-github are mazing!). We have an internal Project board for Papercuts which has a column for 'Needs more investigation / deeper work' (and therefore, those items are less of a priority than some others) but no column yet for 'Never going to happen'. However, if and when we make decisions that something within refined-github is definitely never going to make it into GitHub I'll post here and let y'all know!

**Comment by sibelius** [comment]
@lukehefson any plans to support rendering mdx https://github.com/istarkov/rifm/blob/master/docs/readme.mdx ?

**Comment by lukehefson** [comment]
> any plans to support rendering mdx https://github.com/istarkov/rifm/blob/master/docs/readme.mdx ?

@sibelius I can take a little dig in to see if it's something that might fit in the scope of our initiative!

> I'm always hungry to hear feedback ([tweet me](https://twitter.com/lukehefson/status/1022553956466216960), or @_mention me here)

Side note! I mentioned "@_mention me here", but in order to avoid bloating this thread I probably should have been more specific and said:

"feel free to @_mention me on existing refined-github threads or [tweet me](https://twitter.com/lukehefson/status/1022553956466216960 about new ideas!" 🙌

**Comment by jens1o** [comment]
@lukehefson Open-Source GitHub frontend :P

**Comment by g-plane** [comment]
I wish GitHub can open a new repository then let us propose new ideas and discuss on that repo by using issues.

**Comment by lukehefson** [comment]
Thanks @hkdobrev that is awesome to hear (also… Hi 👋 from a fanboy of your RG contributions😆)

> And while a lot of the features are opinionated or focused on power users, having more than 23k+ installs proves that this could be useful to the wider audience.

Couldn't agree more. I think that it's a pervasive fallacy in software that "you can't satisfy power users _and_ 'regular' users". Of course it's possible! But the problems and solutions need to be well thought out and iterated on – and it's clearly obvious to me (from looking at the history of RG issues and PRs) that the RG community's approach is just this! Bravo!

**Comment by aspiers** [comment]
[**@g-plane**](https://github.com/g-plane) commented on [10 Aug 2018, 00:50 BST](https://github.com/sindresorhus/refined-github/issues/1469#issuecomment-411933034 "2018-08-09T23:50:38Z - Replied by Github Reply Comments"):
> I wish GitHub can open a new repository then let us propose new ideas and discuss on that repo by using issues.

This is exactly what https://github.com/isaacs/github/ is for, but yes it's insane that GitHub don't do this themselves - see isaacs/github#6.

**Comment by Ornataweaver** [comment]
Make #UserStoryHub
Right now only programmers and people around them are making user-stories! Imagine what would happen when everyone would make these stories. ;)

**Comment by lukehefson** [comment]
@sibelius Just to follow up here… I looked into it and mdx render support is probably larger than what we'd consider a 'papercut'. However, the good news is that we already have a process for langauge submissions via the open source [Linguist project](https://github.com/github/linguist)!:

> Before we can perform syntax highlighting for a language, it needs to have an open source Textmate-style language grammar. These grammars also provide syntax highlighting for the Textmate, Sublime Text, and Atom editors:
>
> http://manual.macromates.com/en/language_grammars
>
> Once you've found or created a grammar you can submit it to linguist by following this guide: 
> https://github.com/github/linguist/blob/master/CONTRIBUTING.md

If you've got any additional questions, reach out to me on [twitter](http://twitter.com/lukehefson) or [contact our amazing support folk](http://github.com/contact/).

**Comment by JacobEvelyn** [comment]
@lukehefson great to hear about the new Papercuts team! I'm not sure I totally understand where I should be putting feedback though... in this repo? In https://github.com/isaacs/github? Somewhere else?

For example, if my team and I sorely miss [Bitbucket's git receiving script PR links](https://stackoverflow.com/q/30518603/1103543) (I should note that [GitLab does this too](https://github.com/paulirish/git-open/issues/103#issuecomment-341080201)), where would I put that feedback?

**Comment by lukehefson** [comment]
> in this repo? In https://github.com/isaacs/github? Somewhere else?

@JacobEvelyn this repo and https://github.com/isaacs/github are community-run spaces that we at GitHub read and take a lot of inspiration from (especially on Papercuts!).

However, our official channel for submitting feature requests is via our amazing Support team. You can use either the [contact page](https://github.com/contact), [email](mailto:support@github.com) or [twitter](https://twitter.com/githubhelp).

**Comment by notlmn** [comment]
IMO email, contact page, or twitter are not follow-able (cannot subscribe, no threads), so the Papercuts team should have a separate repo to discuss on. Until then https://github.com/isaacs/github would be the ideal place.

However, this repo was meant to implement features that the community felt nice to have, because any people couldn't ask GH to implement them (in one way, because how email works, and Twitter is not a great place for discussions). And because the current state of this extension is still a doubt (would GH implement suggestions made to this repo, or this extension implement those suggestions and later GH adds them?).

**Comment by pradyunsg** [comment]
I notice the inactivity here so I'll just poke with a request for "Links to an issue's closing commit or pull request" to be considered as a paper cut improvement to GitHub as well.

**Comment by vanniktech** [comment]
@lukehefson will GitHub look into providing some more shortcuts? With the refined GitHub plugin you can already navigate pretty decently with shortcuts only but there are still some things that are missing (e.g. merging PRs, deleting branches and other small things).

**Comment by lukehefson** [comment]
Hi @vanniktech!

> will GitHub look into providing some more shortcuts? 

Yes, I think so. As the papercuts team is so cross-functional, we're cautious of stamping various keyboard shortcuts across lots of different areas of the product – as we need to keep in mind what future roadmaps and plans other team's have. Every time we add a shortcut it means creating a pattern that would be hard to roll back if something else changes in the near future.

However, we're definitely excited about adding more where we can and looking out for great opportunities to add them. [Let me know](https://twitter.com/lukehefson/status/1022553956466216960) if you think there are any in particular that seem low-risk/high-gain to you and I'll see what I can do!
